src/WhiteLibrary/keywords/application.py

```diff
@@ -26,6 +27,25 @@ def launch_application(self, sut_path, args=None):
         else:
             self.state.app = Application.Launch(sut_path)
 
+    def _attach_application(self, sut_identifier):
+        if self.state._attach_application_timeout == 0:
+            return Application.Attach(sut_identifier)
+
+        hack = {"sut": None}
+
+        def search_application():
+            try:
+                hack["sut"] = Application.Attach(sut_identifier)  # noqa: F841
+                return True
+            except WhiteException:
+                return False
+
+        Wait.until_true(search_application,
+                        "{} milliseconds".format(self.state._attach_application_timeout),
```

Comment:
Configuration keywords has this formatter, maybe it should be made into public static method in some helper class ..